packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -1144,6 +1218,36 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         }
     }
 
+    /**
+     * @notice Internal function to claim the boosted reward of the delegator.
+     *
+     * @param validator     Address of the validator.
+     * @param delegator     Address of the delegator.
+     * @param isCompounding Flag to auto compound the boosted reward.
+     *
+     * @return The amount of the claimed boosted reward.
+     */
+    function _claimBoostedReward(
+        address validator,
+        address delegator,
+        bool isCompounding
+    ) internal returns (uint128) {
+        Vault storage vault = _vaults[validator];
+        KghDelegator storage kghDelegator = vault.kghDelegators[delegator];
+
+        uint128 rewardPerKghStored = vault.asset.rewardPerKghStored;
+        uint128 totalBoostedReward = kghDelegator.kghNum *
+            (rewardPerKghStored - kghDelegator.rewardPerKghPaid);
+
+        kghDelegator.rewardPerKghPaid = rewardPerKghStored;
+
+        if (isCompounding) {
+            _delegate(validator, delegator, totalBoostedReward);
+        }
```

Comment:
It seems that this can be handled when calling `_claimBoostedReward()` since `totalBoostedReward` is returned.
If compounding is needed, caller can delegate the `totalBoostedReward` to validator. wdyt?

Reply:
I didn't get what you mean. Could you elaborate it?

Reply:
Ok. I suggest to remove `isCompounding` flag and just return `totalBoostedReward` in this function.
Then, caller can choose to delegate the `totalBoostedReward` or not. 

At first, I thought the compounding delegation should be executed in this function, but it seems to be out of this function. 

For example, in `claimKghReward`, just transfer the `totalBoostedReward` to the delegator. On the other hand, in `delegateKgh` (would be handled in #349), make delegate with the `totalBoostedReward`. Then we don't need the `isCompounding` flang here.

Reply:
I agree it. Just return totalBoostedReward and handle auto compounding out of this function.

Additionally, I suggest to add internal view functions to calculate boosted reward and base reward of KGH delegator each, and use the two functions in `getKghReward` and `claimKghReward`. wdyt?
Then `kghDelegator.rewardPerKghPaid = rewardPerKghStored;` this part should be handled in `claimKghReward` and `delegateKgh`.

Reply:
So you mean that the compounding process will be executed separately in `delegateKgh`? Make sense, I'll remove it!

Reply:
> Additionally, I suggest to add internal view functions to calculate boosted reward and base reward of KGH delegator each, and use the two functions in getKghReward and claimKghReward. wdyt?
Then kghDelegator.rewardPerKghPaid = rewardPerKghStored; this part should be handled in claimKghReward and delegateKgh.

I've also thought about this, but in terms of base reward, making an internal function is not desirable in terms of gas usage, since we have to check `delegateAt` inside the for loop in `claimKghReward`. Also for the boosted reward, `_claimBoostedReward` is also used in `delegateKgh`, which means an internal function to calculate the reward makes `kghDelegator.rewardPerKghPaid = rewardPerKghStored;` line repeated inside `delegateKgh` function, and I didn't see much differences in code readability imo. What do you think?

Reply:
OK, it's not that critical thing and I just suggested it to reduce code redundancy between the `getKghReward` function and `claimKghReward` function.